tests/benchmark/utils/s2n_negotiate_client_benchmark.cc

```diff
@@ -0,0 +1,231 @@
+        print_s2n_error("Error setting host name verification function.");
+    }
+
+    if (type == S2N_STATUS_REQUEST_OCSP) {
+        if (s2n_config_set_check_stapled_ocsp_response(config, 1)) {
+            print_s2n_error(
+                    "OCSP validation is not supported by the linked libCrypto implementation. It cannot be set.");
+        }
+    }
+
+    unsafe_verify_data->trusted_host = host;
+
+    uint8_t mfl_code = 0;
+
+    GUARD_EXIT(s2n_config_send_max_fragment_length(config, (s2n_max_frag_len) mfl_code),
+               "Error setting maximum fragment length");
+
+
+    if (insecure) {
+        GUARD_EXIT(s2n_config_disable_x509_verification(config), "Error disabling X.509 validation");
+    }
+
+    struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+
+    if (conn == NULL) {
+        print_s2n_error("Error getting new connection");
+        exit(1);
+    }
+
+    GUARD_EXIT(s2n_connection_set_config(conn, config), "Error setting configuration");
+
+    GUARD_EXIT(s2n_set_server_name(conn, server_name), "Error setting server name");
+
+    GUARD_EXIT(s2n_connection_set_fd(conn, sockfd), "Error setting file descriptor");
+
+    GUARD_EXIT(s2n_connection_set_client_auth_type(conn, S2N_CERT_AUTH_OPTIONAL),
+               "Error setting ClientAuth optional");
+
+    if (use_corked_io) {
+        GUARD_EXIT(s2n_connection_use_corked_io(conn), "Error setting corked io");
+    }
+
+    GUARD_EXIT(s2n_setup_external_psk_list(conn, psk_optarg_list, psk_list_len), "Error setting external psk list");
+
+    if (benchmark_negotiate(conn, sockfd, state, warmup) != 0) {
+        state.SkipWithError("Negotiate Failed\n");
+        if (DEBUG_PRINT) {
+            printf("Error in negotiate!\n");
+        }
+    }
+
+    if (DEBUG_PRINT) {
+        printf("Connected to %s:%s\n", host, port);
+    }
+
+    GUARD_EXIT(s2n_connection_free_handshake(conn), "Error freeing handshake memory after negotiation");
+
+    s2n_blocked_status blocked;
+    int shutdown_rc = s2n_shutdown(conn, &blocked);
+    if (shutdown_rc == -1 && blocked != S2N_BLOCKED_ON_READ) {
+        fprintf(stderr, "Unexpected error during shutdown: '%s'\n", s2n_strerror(s2n_errno, "NULL"));
+        exit(1);
+    }
+
+    GUARD_EXIT(s2n_connection_free(conn), "Error freeing connection");
+
+    GUARD_EXIT(s2n_config_free(config), "Error freeing configuration");
+
+    free(unsafe_verify_data);
+}
+
+static void ClientBenchmark(benchmark::State& state) {
+    int i;
+    for(i = 0; i < WARMUP_ITERS; i++) {
+        client_benchmark(state, true);
+    }
+    int counter = 0;
+    for (auto _ : state) {
+        counter++;
+        if(DEBUG_PRINT){
+            printf("Count: %d\n", counter);
+        }
+        state.PauseTiming();
+        client_benchmark(state, false);
+    }
+}
+
+int Client::start_benchmark_client(int argc, char** argv) {
+    s2n_init();
+    argument_parse(argc, argv);
+
+    char **newv = (char**)malloc((argc + 4) * sizeof(*newv));
```

Comment:
The argument rewriting seems kinda weird to me. Is there no other way to pass arguments to `Initialize`?

Reply:
Initialize is only able to take in a const char* with the original arguments. Has been now changed to use C++ vectors